clar2wasm/src/words/conditionals.rs

```diff
@@ -213,6 +213,97 @@ impl Word for Filter {
     }
 }
 
+fn traverse_short_circuting_list(
+    generator: &mut WasmGenerator,
+    builder: &mut walrus::InstrSeqBuilder,
+    args: &[SymbolicExpression],
+    invert: bool,
+) -> Result<(), GeneratorError> {
+    let n_branches = args.len();
+
+    let mut branches = vec![];
+
+    let noop = builder
+        .dangling_instr_seq(InstrSeqType::new(
+            &mut generator.module.types,
+            &[],
+            &[ValType::I32],
+        ))
+        // for now, the noop branch just adds a false to break out of the next iteration
+        .i32_const(if invert { 1 } else { 0 })
+        .id();
+
+    for i in 0..n_branches {
+        let branch_expr = args.get_expr(i)?;
+
+        let mut branch = builder.dangling_instr_seq(InstrSeqType::new(
+            &mut generator.module.types,
+            &[],
+            &[ValType::I32],
+        ));
+
+        generator.traverse_expr(&mut branch, branch_expr)?;
+
+        branches.push(branch.id());
+    }
+
+    builder.i32_const(if invert { 0 } else { 1 });
+
+    for branch in branches {
```

Comment:
I like this technique. I think you could simplify it a bit further by creating a block, and adding a `br_if` after each expression to exit the block. I believe this would save some instructions over the if/else solution.

Reply:
I think that would be better too, for now i think this is acceptable and i'd rather continue with the rest of the impls, we can always come back to optimize